detox/e2e/test/messaging/permalink.e2e.js

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// *******************************************************************
+// - [#] indicates a test step (e.g. # Go to a screen)
+// - [*] indicates an assertion (e.g. * Check the title)
+// - Use element testID when selecting an element. Create one if none.
+// *******************************************************************
+
+import {MainSidebar} from '@support/ui/component';
+import {ChannelScreen, PermalinkScreen} from '@support/ui/screen';
+import {User, Setup, Channel, Post} from '@support/server_api';
+
+describe('Permalink', () => {
+    let testUser;
+    let testChannel;
+    let townSquareChannel;
+
+    beforeAll(async () => {
+        const {user, team, channel} = await Setup.apiInit();
+        testUser = user;
+        testChannel = channel;
+        ({channel: townSquareChannel} = await Channel.apiGetChannelByName(team.name, 'town-square'));
+
+        // # Open channel screen
+        await ChannelScreen.open(user);
+    });
+
+    afterAll(async () => {
+        await ChannelScreen.logout();
+    });
+
+    const expectPermalinkTargetMessage = async (permalinkTargetPost) => {
+        // # Post a message in the test channel referencing the given permalink.
+        const permalinkMessage = `[permalink](/_redirect/pl/${permalinkTargetPost.id})`;
```

Comment:
```suggestion
        const permalinkLabel = `permalink-${Date.now().toString()}`;
        const permalinkMessage = `[${permalinkLabel}](/_redirect/pl/${permalinkTargetPost.id})`;
```

Reply:
Thank you for this -- it wasn't obvious until the tests were running together why this wouldn't work as well.